### PR TITLE
Re-apply pulling data from the hub

### PR DESCRIPTION
**Pull request by gboone**
This will reintroduce the changes made by #489 but should not be merged until the following problems are sorted out:

- [x] some team members have incomplete data (no first_name or last_name)
- [x] others have no photo
- [x] README should be updated to point at the go script rather than the `script/` directories

### CODE REVIEW

**Comment by gboone** [comment]
@afeld looks like you didn't break the build so, we're good :smile:

**Comment by gboone** [comment]
The latest few commits here add some modifications to the `author` plugin that includes a filter we can use on the home page like this: `{{ member | with_pic }}`. All that's left to merge this is an update to the main README with updated instructions on getting the site running locally and that (for now) we still need you to submit an image to us if you want your face to show up on the front page.

**Comment by gboone** [comment]
Oh, and because we have the `{{ member | with_pic }}` filter we no longer need to check if a team member is hidden before outputting their name and picture. The filter outputs nothing if Jekyll can't find an image with the same `name` as the team member. This should also prevent us from ever having a name show up on the site without a photo again.

**Comment by gboone** [comment]
So, after following along with all the conversation happening around 18F/Hub yesterday I got to thinking about how much it's worth our time merging this and using Jekyll to pull in the Hub's data.

Is there an advantage to doing all this work with JavaScript that requests data from the Hub's API? This PR adds a bit of complexity mostly in the form of git submodules that need to be updated somehow. From what I can tell the possibilities for keeping the data up to date are 

1. The Series of Webhooks plan where `data-private` fires off a webhook to `staging` that updates the data and commits the resulting `team.yml` and `project.yml` files back to this repo. That, in turn triggers the webhook that builds the site.
2. Individuals and the site team periodically refreshing the data by updating the submodule and running `_data/import_data.rb`
3. Any data pulled out of the Hub should be done through the Hub's JSON API.

Advantage to opt. : Ordinary users don't really need to worry about how the data are getting there because it should always be fresh. The disadvantage is that it seems like a lot of automated complexity and that gives me a mild case of the fantods.

Advantage to opt. 2: The deployment infrastructure stays the same. The disadvantage is the relatively higher barrier of entry for folks who don't want to mess around with submodules, ruby, or who really want to build the site as quickly as possible.

Option three seems to cover both the advantages of the other two without either of their disadvantages.

Discuss.

File: _plugins/author.rb [hunk 1]
```diff
@@ -19,6 +19,18 @@ def render(context)
       end
     end
   end
+
+  module AuthorFilter
+    def with_pic(input)
+      if File.exist?("#{Jekyll.sites[0].config['source']}/assets/images/team/#{input['name']}.jpg")
```
Comment (mbland) [inline _plugins/author.rb]: Pretty sure `Jekyll.sites[0].config['source']` can be replaced with `site.source`.

File: _posts/2014-11-17-taking-control-of-our-website-with-jekyll-and-webhooks.md [hunk 2]
```diff
@@ -27,7 +27,7 @@ It may not look a lot different on the surface, but we made some serious infrast
 
 Deploying our blog posts from GitHub instead of managing a separate Tumblr account has drastically simplified our publishing process and opened it up to the whole team. While we still have an approval process for the _content_ of our blog posts, now anyone on the team [can hit the Publish button](https://github.com/18F/18f.gsa.gov/pull/314).
 
-To that end, we've written a [team blogging guide](https://github.com/18F/18f.gsa.gov/blob/staging/blogging.md) to make the blogging process easier for teammates less familiar with Jekyll.
+To that end, we've written a [team blogging guide](https://github.com/18F/18f.gsa.gov/blob/staging/_posts/blogging.md) to make the blogging process easier for teammates less familiar with Jekyll.
```
Comment (mbland) [inline _posts/2014-11-17-taking-control-of-our-website-with-jekyll-and-webhooks.md]: This should be `README.md` now, shouldn't it?

**Comment by mbland** [comment]
I support whatever you and the team decide, of course. It does seem like option 3 would suffice, if we've got the JavaScript hands on-deck to help out, which it seems we might. So, basically, the Hub does all the data munging and API publishing, so no one else has to. :-) Then it comes down to using JS to generate the page layouts vs. Jekyll+Liquid+Markdown.

File: _plugins/author.rb (hunk 1 above)
Comment (mbland) [inline _plugins/author.rb]: Whoops, this is a filter, not a generator plugin; nevermind.

File: _posts/2014-11-17-taking-control-of-our-website-with-jekyll-and-webhooks.md (hunk 2 above)
Comment (gboone) [inline _posts/2014-11-17-taking-control-of-our-website-with-jekyll-and-webhooks.md]: it should!

**Comment by afeld** [comment]
If the plan is still to [combe the sites](https://github.com/18F/18f.gsa.gov/issues/455), probably best to just go with the easiest thing that works for now.

**Comment by gboone** [comment]
> go with the easiest thing that works for now.
In the interest of doing that, let's merge this now.

**Pull request merged**